### PR TITLE
[WIP] Adding List Domains method to test connection

### PR DESCRIPTION
**Pull request by noiarek**

### CODE REVIEW

File: cadence/gateway/cadence/interface.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from abc import abstractmethod
+from typing import Tuple
+
+from cadence.cadence_types import ListDomainsRequest, ListDomainsResponse
+
+
+class CadenceGatewayInterface:
```
Comment (longquanzheng) [inline cadence/gateway/cadence/interface.py:7]: When it's ready, maybe switch to `WorkflowService` to be backward compatible with the existing one. 
And then the implementation classes can be named `TChannelWorkflowService` and `GrpcWorkflowService`. 
This is fitting our naming pattern in other SDK like Java. 

I feel like `Gateway` is more used as a server concept, this is just a client connecting to server.

File: cadence/gateway/cadence/grpc_gateway.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        self.domain = service_domain_pb2_grpc.DomainAPIStub(self.channel)
+
+    def list_domains(self, request: ListDomainsRequest) -> Tuple[ListDomainsResponse, object]:
+        grpc_request = list_domains_request_to_dataclass(request)
+        response = self.domain.ListDomains.with_call(
+            grpc_request,
+            metadata=self.metadata,
+            timeout=self.timeout
+        )
+        return (list_domains_response_to_dataclass(response[0]), response[1])
```
Comment (longquanzheng) [inline cadence/gateway/cadence/grpc_gateway.py:32]: should it be `proto_list_domains_response_to_dataclass`? or is there some magic that I am not aware of?

File: cadence/gateway/cadence/interface.py (hunk 1 above)
Comment (longquanzheng) [inline cadence/gateway/cadence/interface.py:7]: (please ignore my previous comment).
How about `CadenceServiceInterface`? And having the other two as `CadenceTChannelService` and `CadenceGrpcService`. 
I think gateway is more used as server.

**Review by longquanzheng — APPROVED** [review]
it looks great!

File: cadence/gateway/cadence/grpc_gateway.py (hunk 2 above)
Comment (noiarek) [reply cadence/gateway/cadence/grpc_gateway.py]: It was wrong indeed, the PR was to see if I was in the right direction, it has been fixed :)

File: cadence/gateway/cadence/interface.py (hunk 1 above)
Comment (noiarek) [reply cadence/gateway/cadence/interface.py]: Please feel free to add as many comments like this one as possible. My experience is mainly with servers so I might bring some "bad practices". Fixed

**Pull request merged**